content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+        REDFISHOBJ = server["redfish"]
+        # System Data
+        server_data['System'] = REDFISHOBJ.get(server["ComputerSystem"]).obj
+        # Power Data
+        uri = REDFISHOBJ.get(server["Power"]).obj.Oem.Hpe['Links']['PowerMeter']['@odata.id']
+        server_data["PowerMeter"] = REDFISHOBJ.get(uri).obj
+        # Temperatures
+        server_data['Temperatures'] = REDFISHOBJ.get(server["Thermal"]).obj["Temperatures"]
+        return server_data;
+    
+    except Exception as ex:
+        sys.stderr.write("ERROR during get_server_date: "+server["url"])
+        if ex.status == 401:
+            REDFISHOBJ = redfish.redfish_client(base_url=server["url"], username=login_account, password=login_password)
+            REDFISHOBJ.login()
+            server["redfish"] = REDFISHOBJ
+        log=logopen(lfile)
+        logwriter(log,'Exception')
+        logwriter(log,str(ex.__context__))
+        logclose(log)        
+        pass
+```
+
+## d﻿isplay\_results
+
+The display\_results routine is taking the server\_data retrieved by the get\_server\_data routine and put it into the corresponding Prometheus node gauges.
+
+```python
+def display_results( node, inode, server_metrics, server):
+    hostname = (server_metrics['System']['HostName']).split('.')[0].replace('-','_')
+    cn = server['ilo']
+    inode.labels(cn).info({"Model":server_metrics['System']["Model"],"Manufacturer":server_metrics['System']["Manufacturer"],"SerialNumber":server_metrics['System']["SerialNumber"],"Hostname":hostname})
+    node.labels(cn,server['Rack'],'Power','State').set(power_state[server_metrics['System']["PowerState"]]) 
+    node.labels(cn,server['Rack'],'Power','Average').set(server_metrics["PowerMeter"]['Average'])           
+    node.labels(cn,server['Rack'],'Power','Maximum').set(server_metrics["PowerMeter"]['Maximum'])
+    node.labels(cn,server['Rack'],'Power','Minimum').set(server_metrics["PowerMeter"]['Minimum'])
+    for temperature in server_metrics['Temperatures']:
+        if temperature['Status']['State'] == 'Enabled': 
+            node.labels(cn,server['Rack'],'Temperature',temperature["Name"]).set(temperature['ReadingCelsius'])
+    return 0
+```
+
+The complete most current ILO Prometheus connector (iloPromConnector.v4.3.py at the time of writing this blog) together with a routine (createILOcreadentials.v1.0.py) to generate XML-input file with encrypted user credentials is available on [Github](https://github.com/tbeha/iloPrometheus).
+
+# V﻿isualize the data
+
+The CTC monitor is using Prometheus as a time series database to store and Grafana to visualize the gathered data. Prometheus, Grafana, the  [HPE Storage Array Exporter for Prometheus](https://developer.hpe.com/blog/get-started-with-prometheus-and-grafana-on-docker-with-hpe-storage-array-exporter/) and the [HPE SimpliVity Exporter](https://www.hpe.com/psnow/doc/a50000514enw) are deployed as Kubernetes pods in order to easily scale and adjust the CTC monitoring environment. 
```

Comment:
Change the first part to read "The CTC monitor uses Prometheus..."

Also, do you mean "The CTC monitor" or "The CTC monitoring environment" at the beginning of this paragraph?